bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/AbstractCommerceStorefrontContext.java

```diff
@@ -13,25 +13,55 @@
  ******************************************************************************/
 package com.adobe.cq.commerce.core.components.internal.storefrontcontext;
 
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.caconfig.ConfigurationBuilder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.adobe.cq.commerce.core.components.storefrontcontext.CommerceStorefrontContext;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
-public class AbstractCommerceStorefrontContext implements CommerceStorefrontContext {
+public abstract class AbstractCommerceStorefrontContext implements CommerceStorefrontContext {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractCommerceStorefrontContext.class);
+    private static final String CONFIG_CLASS = "com.adobe.cq.commerce.core.components.internal.storefrontcontext.CommerceStorefrontContextConfig";
+
+    private final Resource resource;
+    private Boolean storefrontContextEnabled;
+
+    public AbstractCommerceStorefrontContext(Resource resource) {
+        this.resource = resource;
+    }
+
+    private boolean isStorefrontContextEnabled() {
+        if (storefrontContextEnabled == null) {
+            storefrontContextEnabled = false;
+            if (resource != null) {
+                ConfigurationBuilder builder = resource.adaptTo(ConfigurationBuilder.class);
+                if (builder != null) {
+                    ValueMap storefrontContextConfig = builder
+                        .name(CONFIG_CLASS).asValueMap();
+                    storefrontContextEnabled = storefrontContextConfig.get("enabled", false);
+                }
+            }
+        }
+
+        return storefrontContextEnabled;
+    }
 
     @Override
     public String getJson() {
-        try {
-            return new ObjectMapper().writeValueAsString(this);
+        if (isStorefrontContextEnabled()) {
```

Comment:
Is there a test that covers the `isStorefrontContextEnabled() = false` case?

Reply:
added now